src/controllers/suggestions.js

```diff
@@ -0,0 +1,385 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+import {
+  badRequest,
+  notFound,
+  ok,
+  createResponse,
+} from '@adobe/spacecat-shared-http-utils';
+import {
+  hasText,
+  isArray,
+  isObject,
+} from '@adobe/spacecat-shared-utils';
+
+import { ValidationError } from '@adobe/spacecat-shared-data-access';
+import { SuggestionDto } from '../dto/suggestion.js';
+
+/**
+ * Suggestions controller.
+ * @param {DataAccess} dataAccess - Data access.
+ * @returns {object} Suggestions controller.
+ * @constructor
+ */
+function SuggestionsController(dataAccess) {
+  if (!isObject(dataAccess)) {
+    throw new Error('Data access required');
+  }
+  const { Suggestion } = dataAccess;
+  if (!isObject(Suggestion)) {
+    throw new Error('Data access required');
+  }
+
+  /**
+   * Gets all suggestions for a given site and opportunity
+   * @param {Object} context of the request
+   * @returns {Promise<Response>} Array of suggestions response.
+   */
+  const getAllForOpportunity = async (context) => {
+    const siteId = context.params?.siteId;
+    const opptyId = context.params?.opportunityId;
+
+    if (!hasText(siteId)) {
+      return badRequest('Site ID required');
+    }
+
+    if (!hasText(opptyId)) {
+      return badRequest('Opportunity ID required');
+    }
+
+    const suggestionEntities = await Suggestion.allByOpportunityId(opptyId);
+    // Check if the opportunity belongs to the site
+    if (suggestionEntities.length > 0) {
+      const oppty = await suggestionEntities[0].getOpportunity();
+      if (!oppty || oppty.getSiteId() !== siteId) {
+        return badRequest('Opportunity not found');
+      }
+    }
+    const suggestions = suggestionEntities.map((sugg) => SuggestionDto.toJSON(sugg));
+    return ok(suggestions);
+  };
+
+  /**
+   * Gets all suggestions for a given site, opportunity and status
+   * @param {Object} context of the request
+   * @returns {Promise<Response>} Array of suggestions response.
+   */
+  const getByStatus = async (context) => {
+    const siteId = context.params?.siteId;
+    const opptyId = context.params?.opportunityId;
+    const status = context.params?.status || undefined;
+    if (!hasText(siteId)) {
+      return badRequest('Site ID required');
+    }
+    if (!hasText(opptyId)) {
+      return badRequest('Opportunity ID required');
+    }
+    if (!hasText(status)) {
+      return badRequest('Status is required');
+    }
+
+    const suggestionEntities = await Suggestion.allByOpportunityIdAndStatus(opptyId, status);
+    // Check if the opportunity belongs to the site
+    if (suggestionEntities.length > 0) {
+      const oppty = await suggestionEntities[0].getOpportunity();
```

Comment:
see above

Reply:
see above :)

Reply:
Agreed to check first for the Opportunity.
Will be done in a separate PR